x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/components/builtin_alert_types/threshold/visualization.tsx

```diff
@@ -226,7 +230,7 @@ export const ThresholdVisualization: React.FunctionComponent<Props> = ({ alert }
     const dateFormatter = (d: number) => {
       return moment(d)
         .tz(timezone)
-        .format(getTimeBuckets(alert.params).getScaledDateFormat());
+        .format(getTimeBuckets(alertParams).getScaledDateFormat());
     };
```

Comment:
Is it possible to avoid `any` and the `as number[][][]` here?

For the `any`: Looking at the underlying type, it looks like it uses `number` explicitly, so we shouldn't need to cast this here. 🤔 

For the `number[][][]`: Looks the core issue is the `any` in `getThresholdAlertVisualizationData`, so can we perhaps validate the result of the API call in `getThresholdAlertVisualizationData` and avoid the `any` there too?